dirty_equals/_datetime.py

```diff
@@ -85,7 +86,11 @@ def prepare(self, other: Any) -> datetime:
             dt = other
         elif isinstance(other, (float, int)):
             if self.unix_number:
-                dt = datetime.fromtimestamp(other)
+                if self.approx is not None and self.approx.tzinfo is not None:
```

Comment:
Should I also require `not self.enforce_tz` in this case? If so, then by default the system timezone will matter, as before, and the tests would need an additional `enforce_tz=False` to pass.

Reply:
I'm not sure about changing the library logic here without a clear reason - this stuff is complicated and it's hard to know what code changes like this might break.

Reply:
OK, then I think also requiring `not self.enforce_tz` is necessary to avoid breaking compatibility. I also think it's sufficient, it's hard to imagine someone being unhappy that `assert some_num == IsDatetime(..., unix_number=True, enforce_tz=False)` started passing instead of failing.

You could also bump the major version and make a breaking change, which might be needed for https://github.com/samuelcolvin/dirty-equals/issues/48.